tests/conftest.py

```diff
@@ -10,3 +19,141 @@
 @pytest.fixture
 def config():
     return CONFIG
+
+
+HEADERS = {
+    RABBITMQ_HEADER_KEY_SUBJECT: "a-subject",
+    RABBITMQ_HEADER_KEY_VERSION: "3",
+}
+
+ENCODED_BODY = "Encoded body"
+VALID_DECODED_LIST = [TEST_CREATE_LABWARE_MSG_OBJECT]
+INVALID_DECODED_LIST = [TEST_INVALID_CREATE_LABWARE_MSG_OBJECT]
+
+
+@pytest.fixture
+def unchecked_create_labware_message():
+    return RabbitMessage(HEADERS, ENCODED_BODY)
+
+
+@pytest.fixture
+def valid_decoder():
+    decoder = MagicMock()
+    decoder.decode.return_value = VALID_DECODED_LIST
+
+    return decoder
+
+
+@pytest.fixture
+def invalid_decoder():
+    decoder = MagicMock()
+    decoder.decode.return_value = INVALID_DECODED_LIST
+
+    return decoder
+
+
+@pytest.fixture
+def valid_create_labware_message(unchecked_create_labware_message, valid_decoder):
+    unchecked_create_labware_message.decode(valid_decoder)
+
+    return unchecked_create_labware_message
+
+
+@pytest.fixture
+def invalid_create_labware_message(unchecked_create_labware_message, invalid_decoder):
+    unchecked_create_labware_message.decode(invalid_decoder)
+
+    return unchecked_create_labware_message
+
+
+@pytest.fixture
+def traction_success_creation_response():
+    return {
+        "data": {
+            "id": "52",
+            "type": "receptions",
+            "links": {"self": "http://localhost:3000/v1/receptions/52"},
+            "attributes": {"source": "traction-ui.sequencescape"},
+        }
+    }
+
+
+@pytest.fixture
+def taxonomy_record():
+    return {
+        "taxId": "9600",
+        "scientificName": "Pongo pygmaeus",
+        "commonName": "Bornean orangutan",
+        "formalName": "true",
+        "rank": "species",
+        "division": "MAM",
+        "lineage": (
+            "Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; "
+            "Mammalia; Eutheria; Euarchontoglires; Primates; Haplorrhini; "
+            "Catarrhini; Hominidae; Pongo; "
+        ),
+        "geneticCode": "1",
+        "mitochondrialGeneticCode": "2",
+        "submittable": "true",
+        "binomial": "true",
+    }
+
+
+@pytest.fixture
+def valid_sample():
+    return {
+        "sampleUuid": "dd490ee5-fd1d-456d-99fd-eb9d3861e0f6".encode(),
```

Comment:
Isn't the strings already in UTF8 ? I think encode() is redundant.

Reply:
I think, I see the reason of the encoding the uuid strings. They are converted to byte arrays.

Reply:
That is to replicate how they would be obtained after unserialising the Avro message